libraries/AP_Networking/AP_Networking_CAN.cpp

```diff
@@ -203,12 +214,27 @@ void AP_Networking_CAN::mcast_server(void)
   handler for CAN frames from the registered callback, sending frames
   out as multicast UDP
  */
-void AP_Networking_CAN::can_frame_callback(uint8_t bus, const AP_HAL::CANFrame &frame)
+void AP_Networking_CAN::can_frame_callback(uint8_t bus, const AP_HAL::CANFrame &frame, AP_HAL::CANIface::CanIOFlags flags)
 {
     if (bus >= HAL_NUM_CAN_IFACES || mcast_sockets[bus] == nullptr) {
         return;
     }
 
+#if AP_NETWORKING_CAN_MCAST_BRIDGING_ENABLED
+    // check if bridged mode is enabled
+    const bool bridged = AP::network().is_can_mcast_ep_bridged(bus);
+#else
+    // never bridge in bootloader, as we can cause loops if multiple
+    // bootloaders with mcast are running on the same network and CAN Bus
+    const bool bridged = false;
+#endif
+
+    if (flags & AP_HAL::CANIface::IsForwardedFrame && !bridged) {
```

Comment:
I think we need parentheses around the bitmask test, or some compilers will complain

Reply:
fixed